fqdn: DNSPoller emits lookups as accesslog/monitor events

raybejjani opened the pull request:
_This is a jankier PR than I would like. There is a bunch of refactoring needed to normalise how DNSPoller, RuleGen and DNSProxy interact with daemon. I didn't want to squeeze that into 1.4._

When relying on monitor events to correlate DNS data to later
connections, information discovered by the poller was lost. These
lookups are now emitted.

```
<- Response dns from 0 ([]) to 0 ([]), identity 0->0, verdict Forwarded DNS Query: google.com. TTL: 273 Answer: '216.58.215.238,2a00:1450:400a:800::200e'
<- Response dns from 0 ([]) to 0 ([]), identity 0->0, verdict Forwarded DNS Query: google.com. TTL: 237 Answer: '216.58.215.238,2a00:1450:400a:800::200e'
```

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/6623)
<!-- Reviewable:end -->

raybejjani committed:
fqdn: DNSPoller emits lookups as accesslog/monitor events

When relying on monitor events to correlate DNS data to later
connections, information discovered by the poller was lost. These
lookups are now emitted.

Signed-off-by: Ray Bejjani <ray@covalent.io>

tgraf commented on daemon/daemon.go:
```diff
@@ -1378,8 +1378,36 @@ func (d *Daemon) bootstrapFQDN(restoredEndpoints *endpointRestoreState) (err err
+			}
+
+			logger.LogTags.Verdict(accesslog.VerdictForwarded, "DNSPoller")(&record)
+			logger.LogTags.DNS(&accesslog.LogRecordDNS{
```
We should add a field to `LogRecordDNS` that will indicate that this is not a DNS lookup made by the app but poller, e.g. `PollerTriggered bool`.

raybejjani replied:
I opted for something a little more general (i.e. a string) so that we can include future sources like api-update or magic-service-discovery-crd, I can just use the boolean, though, if you feel this is a little too much.

raybejjani committed:
accesslog: Add DNS.ObservationSource (poller,proxy)

We want to emit DNS poll responses as monitor events. They fit well
within the current framework but distinguishing intercepted lookups from
endpoints from DNS polls may be difficult. This change adds an
additional field to explicitly indicate where the DNS information is
sourced from, a DNS Proxy or the DNS poller.

Signed-off-by: Ray Bejjani <ray@covalent.io>
